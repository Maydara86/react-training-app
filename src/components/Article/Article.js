import React from 'react'
import styles from './Article.module.css'
import myImage from '../../assets/images/article.jpeg'
import User from '../User/User'
import { ClapFilled, ClapUnfilled, BookmarkFilled, BookmarkUnfilled } from '../Svg/Svg'

export default function Article({
  id,
  articleName,
  claps,
  bookmark,
  responses,
  content,
  user,
  didClap,
  handleClapClick,
  handleBookmarkClick,
  handleStarClick,
  handleFollowClick,
  useArticleMagazinLayout,
}) {
  const magazinTheme = {
    '--article-width': '700px',
    '--article-height': '640px',
    '--article-title-color': 'black',
    '--article-title-fontweight': 'bold',
    '--article-title-size': '30px',
    '--article-border': 'solid',
  }

  const gridTheme = {
    '--article-width': '400px',
    '--article-height': '440px',
    '--article-title-color': 'rgba(0, 0, 0, 0.84)',
    '--article-title-fontweight': '500',
    '--article-border': 'none',
  }

  const theme = useArticleMagazinLayout ? magazinTheme : gridTheme
  Object.keys(theme).map(key => {
    const value = theme[key]
    document.documentElement.style.setProperty(key, value)
  })

  const bookmarkFilled = (
    <BookmarkFilled
      className={styles.bookmarkIconItem}
      key={id}
      bookmarkClick={handleBookmarkClick}
      id={id}
    />
  )
  const bookmarkUnfilled = (
    <BookmarkUnfilled
      className={styles.bookmarkIconItem}
      key={id}
      bookmarkClick={handleBookmarkClick}
      id={id}
    />
  )
  const clapFilled = (
    <ClapFilled className={styles.clapsIconItem} handleClapClick={handleClapClick} id={id} />
  )
  const clapUnfilled = (
    <ClapUnfilled
      className={styles.clapsIconItem}
      handleClapClick={handleClapClick}
      id={id}
      changeSvgColor={useArticleMagazinLayout}
    />
  )

  if (useArticleMagazinLayout) {
    return (
      <div key={id} className={styles.articleContainer}>
        <User
          className={styles.authorItem}
          {...user}
          useSimplifiedLayout
          handleFollowClick={handleFollowClick}
          handleStarClick={handleStarClick}
        />
        <img src={myImage} alt={articleName} className={styles.articleImageItem} />
        <h3 className={styles.articleTitleItem}>{articleName}</h3>
        <p>{content}</p>
        <span className={styles.readMore}>Read more...</span>
        <div className={styles.appreciation}>
          <div className={styles.claps}>
            {didClap ? clapFilled : clapUnfilled}
            <span className={styles.clapsNumber}>{claps}</span>
          </div>
          <div className={styles.responses}>
            <div className={styles.responsesNumber}>{responses} responses</div>
            {bookmark ? bookmarkFilled : bookmarkUnfilled}
          </div>
        </div>
      </div>
    )
  }
  return (
    <div key={id} className={styles.articleContainer}>
      <img src={myImage} alt={articleName} className={styles.articleImageItem} />
      <h3 className={styles.articleTitleItem}>{articleName}</h3>
      <div className={styles.authorAndLikesItem}>
        <div className={styles.authorAndLikesContainer}>
          <User
            className={styles.authorItem}
            {...user}
            useSimplifiedLayout
            handleFollowClick={handleFollowClick}
            handleStarClick={handleStarClick}
          />
          <div className={`${styles.likesItem} ${styles.likesContainer}`}>
            {didClap ? clapFilled : clapUnfilled}
            <div className={styles.clapsNumberItem}>{claps}</div>
            <span className={styles.separatorItem} />
            {bookmark ? bookmarkFilled : bookmarkUnfilled}
          </div>
        </div>
      </div>
    </div>
  )
}
