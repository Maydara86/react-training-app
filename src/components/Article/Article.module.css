h3 {
    font-family: Georgia, "Times New Roman", Times, serif;
    letter-spacing: 0em;
    font-weight: 500;
    line-height: 32px;
    font-size: 24px;
    color: rgba(0, 0, 0, 0.84);
    font-style: normal;
    box-sizing: inherit;
}

h3:hover {
    cursor: pointer;
}


/* ############ ARTICLE CONTAINER ########## */


.articleContainer {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-content: flex-start;
    -ms-flex-line-pack: start;
    align-content: flex-start;
    -webkit-align-items: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    width: var(--article-width);
    height: var(--article-height);
}

.articleImageItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
}

.articleTitleItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    margin-top: 0;
    margin-bottom: 5px;
}

.authorAndLikesItem:nth-child(3) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
}


/* ########### AUTHOR AND LIKES CONTAINER ################ */


.authorAndLikesContainer {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-content: flex-start;
    -ms-flex-line-pack: start;
    align-content: flex-start;
    -webkit-align-items: flex-start;
    -ms-flex-align: start;
    align-items: flex-start;
}

.authorItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
}

.likesItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    margin-top: 10px;
}


/* ########### LIKES CONTAINER ############# */


.likesContainer {
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    -webkit-align-content: flex-start;
    -ms-flex-line-pack: start;
    align-content: flex-start;
    -webkit-align-items: center;
    -ms-flex-align: center;
    align-items: center;
    width: 110px;
}

.clapsIconItem:nth-child(1) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    fill: grey;
    cursor: pointer;
}

.clapsIconItem:hover {
    fill: black;
}

.clapsNumberItem:nth-child(2) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    margin: 3px 0 0 10px;
}

.separatorItem:nth-child(3) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    width: 1px;
    border-right: 1px solid rgba(0, 0, 0, 0.1);
    margin-left: 12px;
    height: 20px;
    margin-right: 10px;
    display: block;
}

.bookmarkIconItem:nth-child(4) {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
    -webkit-align-self: auto;
    -ms-flex-item-align: auto;
    align-self: auto;
    fill: grey;
    cursor: pointer;
}

.bookmarkIconItem:hover {
    fill: black;
}