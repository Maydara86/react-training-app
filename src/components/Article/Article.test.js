import React from 'react';
import {shallow} from '../../enzyme';
import Article, {ClapUnfilled, ClapFilled, BookmarkUnfilled, BookmarkFilled} from './Article'

describe('Article', () => {
    const mockHandleClapClick = jest.fn()
    const mockHandleBookmarkClick = jest.fn()
    const ClapComponent = shallow(<ClapFilled handleClapClick={mockHandleClapClick} />)
    const ArticleComponent = shallow(<Article 
        handleFollowClick={() => {}} 
        handleStarClick={() => {}} 
    />)

    it('renders correctly', () => {
        expect(ArticleComponent).toMatchSnapshot()
    })

    describe('checks bookmark', () => {
        beforeEach(() => {
            ArticleComponent.setProps({bookmark: true})
        })

        it('renders BookmarkFilled component', () => {
            expect(ArticleComponent.find('BookmarkFilled')).toBeTruthy()
        })
    })

    describe('when user did clap', () => {
        beforeEach(() => {
            ArticleComponent.setProps({didClap: true})
        })
        it('renders ClapFilled component', () => {
            expect(ArticleComponent.find('ClapFilled')).toBeTruthy()
        })

        // it('checks the handleClapClick method', () => {
        //     ArticleComponent.find('ClapFilled').simulate('click')
        //     expect(ClapComponent.mockHandleClapClick).toHaveBeenCalled()
        // })
    })
})