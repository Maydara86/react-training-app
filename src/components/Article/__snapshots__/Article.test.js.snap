// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `ShallowWrapper {}`;
